src/locationTracker/LocationGroupHeader.js

```diff
@@ -42,46 +42,21 @@ function LocationGroupHeader(props) {
     });
 
     const displayMenu = useCallback((e) => {
-        show({ event: e, props: { setAllLocationsChecked, setSots, setBarren, setPath } });
+        show({ event: e, props: { setAllLocationsChecked, setHint } });
     });
 
-    useEffect(() => {
-        if (inEffect) {
-            setInEffect(false);
-        } else {
-            setBarren(false);
-            setSots(false);
-            setInEffect(true);
-        }
-    }, [pathIndex]);
-
-    useEffect(() => {
-        if (inEffect) {
-            setInEffect(false);
-        } else {
-            setBarren(false);
-            setPath(6);
-            setInEffect(true);
-        }
-    }, [sots]);
-
-    useEffect(() => {
-        if (inEffect) {
-            setInEffect(false);
-        } else {
-            setSots(false);
-            setPath(6);
-            setInEffect(true);
-        }
-    }, [barren]);
+    let hint = '';
+    if (props.logic.regionHints !== undefined) {
+        hint = props.logic.regionHints[props.title];
+    };
 
     let image;
-    if (pathIndex < 6) {
-        image = <img src={pathImages[pathIndex]} alt="path" />;
-    } else if (sots) {
-        image = <img src={sotsImage} alt="sots" />;
-    } else if (barren) {
-        image = <img src={barrenImage} alt="barren" />;
+    if (hint.includes('Path')) {
+        image = <img src={pathImages[hint.slice(8)]} alt={hint} />;
+    } else if (hint === 'Spirit of the Sword') {
+        image = <img src={sotsImage} alt={hint} />;
+    } else if (hint === 'Barren') {
+        image = <img src={barrenImage} alt={hint} />;
```

Comment:
Personally I'd use something more structured in the data - e.g.
```ts
export type Hint =
    | { type: 'barren' }
    | { type: 'sots' }
    | { type: 'path'; index: number }; // or path: string
```
with a function that decodes these hint types for UI.